lms/djangoapps/grades/subsection_grade_factory.py

```diff
@@ -107,8 +107,9 @@ def update(self, subsection, only_if_higher=None, score_deleted=False, force_upd
             self._update_saved_subsection_grade(subsection.location, grade_model)
 
             if settings.FEATURES.get('ENABLE_COURSE_ASSESSMENT_GRADE_CHANGE_SIGNAL'):
```

Comment:
not related to your changes but I tracked this one line where this factory's update method is called, even during a read? wth!!!! https://github.com/edx/edx-platform/blob/9cf2f9f298e5e8be3b3abcaadaf0b7a96d0de0df/lms/djangoapps/grades/course_grade.py#L288

Reply:
yup! haha, we update the subsection quite often, which is why we have the caching and other checks.